lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -180,6 +193,48 @@ circt::rtl::RTLExternModuleOp ESIRTLBuilder::declareStage() {
   return declaredStage;
 }
 
+/// Return the InterfaceType which corresponds to an ESI port type. If it
+/// doesn't exist in the cache, build the InterfaceOp and the corresponding
+/// type.
+InterfaceOp ESIRTLBuilder::getOrConstructInterface(ChannelPort t) {
+  auto ifaceIter = portTypeLookup.find(t);
+  if (ifaceIter != portTypeLookup.end())
+    return ifaceIter->second;
+  auto iface = constructInterface(t);
+  portTypeLookup[t] = iface;
+  return iface;
+}
+
+InterfaceOp ESIRTLBuilder::constructInterface(ChannelPort chan) {
+  auto ctxt = getContext();
+
+  InterfaceOp iface = create<InterfaceOp>(loc, "IDataVR");
+  OpBuilder ib(iface.getRegion());
+  ib.createBlock(&iface.getRegion());
+
+  InterfaceSignalOp s;
+  ib.create<InterfaceSignalOp>(loc, valid.getRootReference(),
+                               TypeAttr::get(getI1Type()));
```

Comment:
This pattern looks a bit weird to me, would it make sense to add a build method to InterfaceSignalOp that takes a type instead of attribute?  I'm not sure the semantics here, but this seems boilerplate-y